sklearn/inspection/_partial_dependence.py

```diff
@@ -81,30 +86,59 @@ def _grid_from_X(X, percentiles, grid_resolution):
 
     values = []
     for feature in range(X.shape[1]):
```

Comment:
It would make sense to rename this as `feature_idx` since it's not the feature name.